src/components/organisms/Common/AssetModal/hooks.ts

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useState, useEffect } from "react";
+import { useIntl } from "react-intl";
+
+import {
+  AssetsQuery,
+  useAssetsQuery,
+  useCreateAssetMutation,
+  useRemoveAssetMutation,
+  Maybe,
+} from "@reearth/gql";
+import { useNotification } from "@reearth/state";
+
+export type AssetNodes = NonNullable<AssetsQuery["assets"]["nodes"][number]>[];
+
+export enum AssetSortType {
+  Date = "DATE",
+  Name = "NAME",
+  Size = "SIZE",
+}
+
+export default (teamId?: string) => {
+  const intl = useIntl();
+  const [, setNotification] = useNotification();
+  const [sort, setSort] = useState<{ type?: Maybe<AssetSortType>; reverse?: boolean }>();
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const assetsPerPage = 20;
+
+  const pagination = useCallback(
+    (endCursor?: string | null) => {
+      const reverseOrder =
+        (!sort?.type && !sort?.reverse) ||
+        (sort?.type === "DATE" && !sort?.reverse) ||
+        (sort?.type !== "DATE" && sort?.reverse);
+
+      return {
+        after: !reverseOrder ? endCursor : undefined,
+        before: reverseOrder ? endCursor : undefined,
+        first: !reverseOrder ? assetsPerPage : undefined,
+        last: reverseOrder ? assetsPerPage : undefined,
+      };
+    },
+    [sort],
+  );
+
+  const { data, refetch, loading, fetchMore, networkStatus } = useAssetsQuery({
```

Comment:
If you close an asset modal and then reopen it, does it fetch data from the server again? If not, the cache has been saved and you should review the fetchPolicy.

Reply:
Should I delete the cache on close? Is there any reason we want a fresh AssetModal each time instead of leaving it as it was until the user changes sort, reverses or searches a word?
Isn't it more performant to not needlessly fetch if they have already been fetched?

Reply:
The assumption is that the asset should not remain cached when the dialog is closed. The more paging is done, the more memory is consumed, and once the dialog is closed, the asset is no longer used, so the cache should be deleted when the component is unmounted.

And, since sorting or filtering of assets is done in AssetModal, there should be no concern as long as this component is mounted, the cache will persist even if the fetchPolicy is network-only. If the component is destroyed, the cache will be destroyed.

Please let me know if this perception is incorrect and actually works differently.

Reply:
I figured out I can import useApolloClient and then, accessing the cache through there, evict the fieldName "assets" on close!